jupyter_server_fileid/manager.py

```diff
@@ -36,7 +37,34 @@ def wrapped(self, *args, **kwargs):
     return decorator
 
 
-class FileIdManager(LoggingConfigurable):
+class AbstractFileIdManager(ABC):
```

Comment:
Seems like the `root_dir` attribute should reside here and not be configurable since it will always derive from the server-app/contents manager and we shouldn't assume its value only pertains to filesystem-based contents managers (or the `LocalFileIdManager`).

Should we also say the same for the logic that processes events emitted from the ContentsManager?  Seems like  these would be useful/necessary for all implementations - certainly more than just `LocalFileIdManager`.  Hmm.  On that note, I could see which events are interesting to a given FID implementation could vary.  For example, FIleId managers that are not filesystem-specific will need to listen to content `GET` events - which the `LocalFileIdManager` does not currently do.  So a good abstract method to implement would be one that returns a dictionary of event action handlers for that FID impl.

Reply:
> Seems like the root_dir attribute should reside here and not be configurable since it will always derive from the server-app/contents manager

I agree that this shouldn't be a configurable trait and instead just be a instance attribute. However, this should be passed to `__init__()` from the extension entry point, as a file ID manager should be able to be instantiated outside of a ServerApp context for testing (i.e. the class itself doesn't require `self.settings["contents_manager"]` to be defined).

> a good abstract method to implement would be one that returns a dictionary of event action handlers for that FID impl.

I also agree with this. This should be part of the FileIdManager and not hard-coded in extension setup.

Reply:
This is fixed in the latest revision. Thanks Kevin!